projectmainpage.html

```diff
@@ -0,0 +1,265 @@
+            </ul>
+        </nav>   
+        
+        <!-- loading popup -->
+        <div id="loading-popup">Loading recommendations, please wait 12 seconds...</div>
+
+        <!-- search bar -->
+        <div id="search-container">
+            <input type="text" id="search-bar" placeholder="Search for any products..." />
+        </div>
+
+        <main>
+            <!-- filter section -->
+            <aside id="filter-container">
+                <h3>Product Filters:</h3>
+                <ul>
+                    <li><input type="checkbox" id="foundation" name="foundation"> Foundation</li>
+                    <li><input type="checkbox" id="concealer" name="concealer"> Concealer</li>
+                    <li><input type="checkbox" id="bronzer" name="bronzer"> Bronzer</li>
+                    <li><input type="checkbox" id="lipstick" name="lipstick"> Lipstick</li>
+                    <li><input type="checkbox" id="nail-polish" name="nail_polish"> Nail Polish</li>
+                    <li><input type="checkbox" id="blush" name="blush"> Blush</li>
+                    <li><input type="checkbox" id="powder" name="powder"> Powder</li>
+                </ul>
+                <br>
+                <h3>Skin Tone Filters:</h3>
+                <ul>
+                    **This filter can ONLY be applied when foundation, concealer, and bronzer products are filtered**
+                </ul>
+                    <ul>
+                        <li><input type="radio" id="skin-tone-fair" name="skin-tone" value="fair"> Light</li>
+                        <li><input type="radio" id="skin-tone-medium" name="skin-tone" value="medium"> Medium</li>
+                        <li><input type="radio" id="skin-tone-dark" name="skin-tone" value="dark"> Dark</li>
+                        <li><input type="radio" id="skin-tone-none" name="skin-tone" value="none" checked> No Filter Applied</li>
+                    </ul>
+                <br>
+                <h3>Price Filters:</h3>
+                <ul>
+                    <li><input type="checkbox" id="under-10" name="under-10"> Under $10</li>
+                    <li><input type="checkbox" id="10-20" name="10-20"> $10 - $20</li>
+                    <li><input type="checkbox" id="20-50" name="20-50"> $20 - $50</li>
+                    <li><input type="checkbox" id="50-plus" name="50-plus"> $50 and above</li>
+                </ul>
+            </aside>
+
+            <!-- recommendations section with headers-->
+            <section id="home-recommendations">
+                <h2>Welcome to Makeup Manager!</h2>
+                <br>
+                <br>
+                <h2>Makeup reccomendations for you: </h2>
+                <br>
+                <br>
+                <div id="recommendation-grid"></div>
+            </section>
+        </main>
+
+        <br>
+        <!-- Footer -->
+        <footer id="footer">
+            <p>Copyright by MakeupManager</p>
+        </footer>
+
+        <script>
```

Comment:
When this script block runs, it bricks my computer... That is bad...

Reply:
Fixed.

Used the "head" of each query (first 20), ensuring not-too-many elements on the page. Future implementation could incoporate a global query index for memoizing the next index in the query.